src/thread.ts

```diff
@@ -0,0 +1,115 @@
+import { StreamChat } from './client';
+import {
+  DefaultGenerics,
+  ExtendableGenerics,
+  MessageResponse,
+  ThreadResponse,
+  ChannelResponse,
+  FormatMessageResponse,
+  ReactionResponse,
+  UserResponse,
+} from './types';
+import { addToMessageList, formatMessage } from './utils';
+
+type ThreadReadStatus<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics> = Record<
+  string,
+  {
+    last_read: Date;
+    last_read_message_id: string;
+    unread_messages: number;
+    user: UserResponse<StreamChatGenerics>;
+  }
+>;
+export class Thread<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics> {
+  id: string;
+  latestReplies: FormatMessageResponse<StreamChatGenerics>[] = [];
+  participants: ThreadResponse['thread_participants'] = [];
+  message: FormatMessageResponse<StreamChatGenerics>;
```

Comment:
Is this the thread parent message?

Reply:
yes

Reply:
don't we want to call it parentMessage as the thread reply contains parent_id?

Reply:
I actually called it `parentMessage` before, but somehow felt it easier to just call it a message. I mean it is assumed that thread will have 1 message and multiple replies.

But feel free to change it later during sdk integration if everyone agrees on it.